src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.Android.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+
+namespace System.Net.Http
+{
+    public partial class HttpClient
+    {
+        private const string HandlerSwitchName = "System.Net.Http.UseNativeHttpHandler";
+
+        private static HttpMessageHandler CreateDefaultHandler()
+        {
+            // Default is to use the Android native handler
+            if (!AppContext.TryGetSwitch(HandlerSwitchName, out bool isEnabled))
+            {
+                return new HttpClientHandler();
+            }
+
+            Type? androidEnv = Type.GetType("Android.Runtime.AndroidEnvironment, Mono.Android");
+            return (HttpMessageHandler)androidEnv!.GetMethod("GetHttpMessageHandler", BindingFlags.Public | BindingFlags.Static)!.Invoke(null, null)!;
```

Comment:
The method is not public, the MethodInfo needs to be cached and not to do look-up every time.

(same for other handlers)